test/conftest.py

```diff
@@ -83,6 +84,24 @@ def kill(self, force=True, immediate=True):
         if not force and self.pg.poll() is None:
             raise Exception("PG pid {} not dead".format(self.pg.pid))
 
+    def connect(self):
+        connection_string = pgutil.create_connection_string(self.user)
+        conn = psycopg2.connect(connection_string)
+        return conn
+
+    def switch_wal(self):
+        if self._connection is None:
+            self._connection = self.connect()
+            self._connection.autocommit = True
+        cur = self._connection.cursor()
```

Comment:
It's a bit strange to have both a `connect()`function that returns a new connection every time, and this function that keeps a never-closed connection, do we really need to keep it open and store it ?

Reply:
For the first part: the connect function imo needs to return a new connection every time, as the caller in the test must be able to open two in parallel if needed.

As for this one, I figured it would be better to only keep one single connection to avoid reopening it every single time. That's not necessary but I don't see how it would hurt to keep a connection open for repeated testing operations ? I can remove the caching if you prefer, or move that method to a function that takes the connection as a parameter.

Reply:
> move that method to a function that takes the connection as a parameter

I think that's the best, it keeps the class smaller and avoid the inevitable flaky test we'll have one day because of a connection that's still open.

Reply:
Done.